app/webhook.go

```diff
@@ -789,10 +789,32 @@ func (a *App) HandleIncomingWebhook(c *request.Context, hookID string, req *mode
 		overrideIconURL = req.IconURL
 	}
 
-	_, err := a.CreateWebhookPost(c, hook.UserId, channel, text, overrideUsername, overrideIconURL, req.IconEmoji, req.Props, webhookType, "")
+	postingUserID, getBotUserIdErr := a.GetBotUserId(hook.UserId, req.Username)
+	if getBotUserIdErr != nil {
+		a.Log().Error("Error getting bot user: UserId="+hook.UserId+", Username="+req.Username+".", mlog.Err(getBotUserIdErr))
+	}
+	if postingUserID == "" {
+		postingUserID = hook.UserId
+	}
+
+	_, err := a.CreateWebhookPost(c, postingUserID, channel, text, overrideUsername, overrideIconURL, req.IconEmoji, req.Props, webhookType, "")
 	return err
 }
 
+func (a *App) GetBotUserId(userID string, botUsername string) (string, error) {
+	bots, err := a.Srv().Store().Bot().GetAll(&model.BotGetOptions{Page: 0, PerPage: 10})
+	if err != nil {
+		return "", err
+	}
+	for i := range bots {
+		bot := bots[i]
+		if bot.OwnerId == userID && bot.Username == botUsername {
```

Comment:
Based on how this function is called, this is saying "give me a bot owned by the user that owns the webhook", as well as "search for the bot's username using the provided `username` in the webhook event". We want to be able to _not_ provide the username for the individual incoming events.

Thinking about this some more, I think we can add an optional `poster_id` to the `IncomingWebhook` struct to set to a bot account. This would be assigned on webhook creation, rather than webhook _event_ creation. Does this make sense @jakubriegel? Maybe there's a better name than that though, something that conveys "this user _will_ create a post for any events that come through"

Reply:
It makes sense, but it think it is not in common with the concept of web hooks. For me, here the idea here was to have quick and easy way of posting to Mattermost as a specified user. I far as I understand the API, there are other endpoints for more sophisticated posting in the name of bot

Reply:
@hanzei Can you give this PR and discussion a review when you have the chance?

Reply:
The configuration of a bot account via webhook payload sounds brittle and confusing for webhook users. Is it reasonable to store the "connected" bot account in the webhook struct and configure it via the webapp?

Reply:
@hanzei So you're proposing we "lock" the bot account to the given saved webhook? I'm on board with this direction

Reply:
@mickmister Exactly.

Reply:
@jakubriegel What do you think about this direction?

Reply:
Makes sense 👍🏻 But I may not be able to take the web app part. Would you guys be able to take it from here?